GVFS/GVFS.Platform.POSIX/POSIXFileSystem.cs

```diff
@@ -1,202 +1,202 @@
-
-            // #define S_IXUSR     0000100     /* [XSI] X for owner */
-            private static readonly ushort IXUSR = Convert.ToUInt16("100", 8);
-
-            // #define S_IXGRP     0000010     /* [XSI] X for group */
-            private static readonly ushort IXGRP = Convert.ToUInt16("10", 8);
-
-            // #define S_IXOTH     0000001     /* [XSI] X for other */
-            private static readonly ushort IXOTH = Convert.ToUInt16("1", 8);
-
-            public static bool IsSock(ushort mode)
-            {
-                // #define  S_ISSOCK(m) (((m) & S_IFMT) == S_IFSOCK)    /* socket */
-                return (mode & IFMT) == IFSOCK;
-            }
-
-            public static bool IsExecutable(ushort mode)
-            {
-                return (mode & (IXUSR | IXGRP | IXOTH)) != 0;
-            }
-
-            [DllImport("libc", EntryPoint = "stat$INODE64", SetLastError = true)]
-            public static extern int Stat(string path, [Out] out StatBuffer statBuffer);
-
-            [StructLayout(LayoutKind.Sequential)]
-            public struct TimeSpec
-            {
-                public long Sec;
-                public long Nsec;
-            }
-
-            [StructLayout(LayoutKind.Sequential)]
-            public struct StatBuffer
-            {
-                public int Dev;              /* ID of device containing file */
-                public ushort Mode;          /* Mode of file (see below) */
-                public ushort NLink;         /* Number of hard links */
-                public ulong Ino;            /* File serial number */
-                public uint UID;             /* User ID of the file */
-                public uint GID;             /* Group ID of the file */
-                public int RDev;             /* Device ID */
-
-                public TimeSpec ATimespec;     /* time of last access */
-                public TimeSpec MTimespec;     /* time of last data modification */
-                public TimeSpec CTimespec;     /* time of last status change */
-                public TimeSpec BirthTimespec; /* time of file creation(birth) */
-
-                public long Size;          /* file size, in bytes */
-                public long Blocks;        /* blocks allocated for file */
-                public int BlkSize;        /* optimal blocksize for I/O */
-                public uint Glags;         /* user defined flags for file */
-                public uint Gen;           /* file generation number */
-                public int LSpare;         /* RESERVED: DO NOT USE! */
-
-                [MarshalAs(UnmanagedType.ByValArray, SizeConst = 2)]
-                public long[] QSpare;     /* RESERVED: DO NOT USE! */
-            }
-        }
-
-        private class NativeFileReader
-        {
-            public const int ReadOnly = 0x0000;
-            public const int WriteOnly = 0x0001;
-
-            public const int Create = 0x0200;

```

Comment:
We have the same problem here as in the file-lock code; this is `0x0040` on Linux, and the various `chmod`-related calls should use an `uint`.  There may be other Linux vs. BSD changes here as well, so this probably needs to be refactored into a POSIX base with Linux and Mac extensions.

Reply:
>There may be other Linux vs. BSD changes here as well, so this probably needs to be refactored into a POSIX base with Linux and Mac extensions.

@kivikakk, @chrisd8088, are you planning to make this change in this PR, or in a follow up PR?

Reply:
@kivikakk is away for a bit and as this is their work, I'm not looking to make any major changes to it right now.  My own feeling at the moment is summarized by in my [review comment](https://github.com/Microsoft/VFSForGit/pull/926#discussion_r268354587):

> To that end, the #810 PR (still a WIP) has a number of these modifications in it now, so using that as a basis for refactoring (maybe refactoring after that code has landed?) might be useful.

But I'll defer to @kivikakk to decide if they want to try to get the POSIX changes in first.  For my part, I'm focused on looking at blockers to running `gvfs mount` on Linux, which is somewhat orthogonal work to this refactoring.

Reply:
@chrisd8088 ok thanks!

Regarding the status of this PR, I think it's close being ready to merge, there are just a few open comments left from the first round of review.

/cc: @jrbriggs @kivikakk

Reply:
I'd like to perform these changes later when we add the Linux platform in -- at that point we should refactor so that e.g. there's only one definition of any given native method for a given platform, whereas right now we have `open` and `O_CREAT` defined twice (in FileSystem and FileBasedLock).

It's pretty hard to get a refactor right when there's only one consumer of the code, and right now that's the case (Mac platform code is the single consumer of the POSIX platform base code). We'll maybe find more cases like this where Linux varies vs. what we've pre-emptively abstracted, but we should handle them as we implement the Linux platform in the tree, and not try to get this PR 100% correct before (imo). More important is that we get the initial refactor down asap so we don't get increasingly large deltas as time presses on.

Reply:
Sounds like a reasonable plan!